### ISSUE TITLE
tiddlydesktop constantly sends outgoing connections

### ISSUE THREAD

**Issue by astro46** [Jermolene/TiddlyDesktop#206]
just installed,  I am finding that it is constantly sending outgoing connections, sometimes to google (eg: 172.217.8.170 or 172.217.6.10 ) other times to ? (eg: 239.255.255.250) or other addresses.

What is going on?

**Comment by 5agado** [comment]
I also noticed the *nw.exe* process sending and receiving bytes in the network activity monitor.

Looking for explanations or ways to inspect what it is up to.

**Comment by Jermolene** [comment]
Hi @astro46 @5agado the only outgoing network connections made by TiddlyDesktop itself are when one chooses a HTTP-hosted template for a new wiki. It's possible you're seeing traffic generated by nw.js itself, or rather from the embedded copy of Chromium. The nw.js maintainers do sometimes have to play whack-a-mole with unexpected network activity (for example, [this issue](https://github.com/nwjs/nw.js/issues/5865)).

I installed Little Snitch on my Mac to see if I could duplicate the problem, but I didn't see any alerts until I created a new wiki from an HTTP-hosted template.

What OS are you using?

**Comment by astro46** [comment]
I win10 pro.  I am using a tiddlywiki that I downloaded last week and have been experimenting with.  It is stored on my computer.   the network traffic is from nw.js.   some of the url's were to google, but there were other addresses that I didn't investigate, as noted in my original post.

**Comment by 5agado** [comment]
Win10 Pro here too.
I didn't use any HTTP-hosted template. I just got a bit paranoid that maybe some of the plugins might be doing something nasty. But I only got official Markdow parser + KaTex. Plus [Encrypt Tiddler](https://tiddlywiki.com/static/%2522Encrypt%2520single%2520tiddler%2520plugin%2522%2520by%2520Danielo%2520Rodriguez.html) and [Highlight Searched Text](https://github.com/bimlas/tw5-highlight-searched-text).
In the doubt, I added a Windows firewall rule to block and in and out connections for the *nw.exe* file. I'll double check on my Mac setup tomorrow too.

**Comment by astro46** [comment]
right.  I considered blocking nw.exe, but since it seemed to be going to numerous different addresses, I didn't know if there was any value to me in it connecting with  any of them.  even if total blocking is ok, I don't need a program using computer time repeatedly trying to access out.  there needs to be a way to turn it off.

**Comment by 5agado** [comment]
I confirm that the same is happening on MacOS. Not sure how TiddlyDesktop works, but I also tried to remove any TiddlyWiki file and reopon it with an empty list, and I still see the activity going on.

Here a screenshot from Little Snitch

<img width="310" alt="Screenshot 2020-07-16 at 15 27 23" src="https://user-images.githubusercontent.com/1591484/87688416-ddd8c880-c77e-11ea-929a-35972a9594da.png">

**Comment by 5agado** [comment]
As an update, I tried with a fresh new installation (v0.0.14), and at the first run many domains are called: tiddlywiki.com, gstatic.com, googleapis.com and finally the unknown address pointed in the screenshot above at port 53

**Comment by astro46** [comment]
creepy.  Slovakia??  Wanting to connect with google is bad(and unacceptable) enough.

**Comment by Jermolene** [comment]
Hi @astro46 @5agado I need to investigate further: I installed Little Snitch on my Mac but as I noted above I couldn't see any unexpected network activity. However, I'm not a regular user of Little Snitch and it's possible that I wasn't using it correctly.

I think the traffic you're reporting is consistent with my suggestion that the cause is the embedded Chromium within nw.js, and I'll investigate further.

Any further observations are appreciated.

>  I guess is expected for TiddlyDesktop not to have a code signature.

It's just a question of time and effort. Any help is welcome on this and other areas.

**Comment by BurningTreeC** [comment]
maybe [this nw.js issue](https://github.com/nwjs/nw.js/issues/5865) can help

[here](https://github.com/nwjs/nw.js/issues/5865#issuecomment-585225440) someone is suggesting some arguments to pass to nw

**Comment by 5agado** [comment]
@Jermolene appreciate the effort.

I am not expert with Little Snitch either, but I would suggest to run in full alert mode (will request permission for any connection). Is a bit of a pain if you have open browsers or such, as it will trigger a prompt at every attempt, but otherwise should exactly show up anything tried by TiddlyDesktop.

Given that I downloaded a new TiddlyDesktop version I doubt that any of such connections can be related to my setup, but good if someone else can also confirm these findings.

**Comment by astro46** [comment]
On a win10 machine.  I use Windows Firewall Notifier,  which alerts about any outgoing request for which there isn't an existing rule.  I can  create an allow or block rule,  create a temporary rule, or ignore (which denies the request at the moment with no rule established.  some requests gets repeated notifications going for different ports. I've seen progs getting into the hundreds. they just can't accept defeat).  when running TD there are a series of alerts in the course of using the program, as i mentioned above.

**Comment by pdroalves** [comment]
Any update on this?

**Comment by Jermolene** [comment]
Hi @pdroalves apologies, I'm afraid I've not got beyond failing to get Little Snitch to work to be able to research this further.

I'm planning to work on a new version of TiddlyDesktop once the current TiddlyWiki 5 release is out of the way in the next couple of weeks, and will look at it then.
